c/src/ml-api-service-remote.c

```diff
@@ -575,70 +578,81 @@ ml_service_remote_release_internal (ml_service_s * mls)
   return ML_ERROR_NONE;
 }
 
+/**
+ * @brief Set path in ml-service remote handle.
+ */
+int
+ml_service_remote_set_path (ml_service_h handle, const gchar * path)
+{
+  ml_service_s *mls = (ml_service_s *) handle;
+  _ml_remote_service_s *mlrs = (_ml_remote_service_s *) mls->priv;
+
+  if (!g_file_test (path, G_FILE_TEST_IS_DIR)) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The given param, dir path = \"%s\" is invalid or the dir is not found or accessible.",
+        path);
+  }
+  if (g_access (path, W_OK) != 0) {
+    _ml_error_report_return (ML_ERROR_PERMISSION_DENIED,
+        "Write permission denied, path: %s", path);
+  }
+
+  g_free (mlrs->path);
+  mlrs->path = g_strdup (path);
+
+  return ML_ERROR_NONE;
+}
+
 /**
  * @brief Creates ml-service handle with given ml-option handle.
  */
 int
-ml_service_remote_create (ml_option_h option, ml_service_event_cb cb,
-    void *user_data, ml_service_h * handle)
+ml_service_remote_create (ml_service_h handle, ml_option_h option)
 {
   ml_service_s *mls;
   _ml_remote_service_s *remote_s = NULL;
   edge_info_s *edge_info = NULL;
   int ret = ML_ERROR_NONE;
-  gchar *_path = NULL;
+  gchar *_path;
```

Comment:
It seems there is no problem in logic to remove NULL declaration.
Are there any reason to remove this? (Just curious)

Reply:
This is my mistake. I reverted this change. Thanks.